jwala-services/src/main/java/com/cerner/jwala/service/jvm/impl/JvmServiceImpl.java

```diff
@@ -704,7 +721,67 @@ protected void deployScriptsToUserJwalaScriptsDir(Jvm jvm, User user) throws Com
         }
 
     }
-
+    
+   /**
+    * This method upgrades the JDK version for the mentioned JVM
+    * 
+    */
+	public Jvm upgradeJDKAndDeployJvm(String jvmName, User user) {
+		boolean didSucceed = false;
+		Jvm jvm = getJvm(jvmName);
+		LOGGER.info("Start upgradeJDKAndDeployJvm for {} by user {}", jvmName, user.getId());
+
+		historyFacadeService.write(jvm.getHostName(), jvm.getGroups(),
+				"Starting to upgrade JDK :: JVM " + jvm.getJvmName() + " on host " + jvm.getHostName(),
+				EventType.USER_ACTION_INFO, user.getId());
+
+		// add write lock for multiple write
+		binaryDistributionLockManager.writeLock(jvmName + "-" + jvm.getId().toString());
+
+		try {
+			// Validation 1: Application should be in STOPPED state
+			if (!jvm.getState().equals(JvmState.JVM_STOPPED)) {
+				final String errorMessage = "The remote JVM ::" + jvm.getJvmName()
+						+ " must be in STOPPED state before attempting to upgrade the JDK of the JVM";
+				LOGGER.error(errorMessage);
+				throw new InternalErrorException(FaultType.REMOTE_COMMAND_FAILURE, errorMessage);
+			}
+
+			// Step 1: Check if JDK Binaries exists
+			checkForJvmBinaries(jvm);
+			// Step 2: Distribute JDK binaries if required
+			distributeBinaries(jvm);
+
+			// Step 3: delete the service
+			deleteJvmService(jvm, user);
+
+			// Step 4: update the Setenv file with upgraded JDK
+			deployJvmResourceFilesForJDKUpgrade(jvm, user);
+
+			// Step 5: re-install the service
+			installJvmWindowsService(jvm, user);
+
+			didSucceed = true;
+		} catch (CommandFailureException | IOException e) {
+			LOGGER.error("Failed to upgrade JDK :: the JVM config for {}", jvm.getJvmName(), e);
+			throw new InternalErrorException(FaultType.REMOTE_COMMAND_FAILURE,
+					"Failed to generate the JVM config: " + jvm.getJvmName(), e);
+		} finally {
+			binaryDistributionLockManager.writeUnlock(jvmName + "-" + jvm.getId().toString());
+			LOGGER.info("End upgrade JDK for {} by user {}", jvmName, user.getId());
+
+			final EventType eventType = didSucceed ? EventType.SYSTEM_INFO : EventType.SYSTEM_ERROR;
+
+			String historyMessage = didSucceed

```

Comment:
When the method succeeds add the JDK version that was installed to the history message as well.